resolver.go

```diff
@@ -140,38 +142,67 @@ func (r *resolver) Start() error {
 	if r.err != nil {
 		return r.err
 	}
+	// startCh is to seralize resolver Start and Stop
+	r.Lock()
+	r.startCh = make(chan struct{})
```

Comment:
`NewResolver()` is called once for a sandbox. But for a given resolver `Start` and `Stop` can happen more than once; ie: once per container restart. The intent of the change is to serialize a  pair of `Start` and `Stop` calls. That is the reason for not creating `startCh` in the constructor.